app/classifier/subject-viewer.cjsx

```diff
@@ -88,6 +90,26 @@ module.exports = React.createClass
               </g>}
         </svg>
 
+        {if @props.subject.already_seen or @props.subject.id in sessionSubjects
+          <button type="button" className="warning-banner" onClick={@toggleWarning}>
+            Already seen!
+            {if @state.showWarning
+              <Tooltip attachment="top left" targetAttachment="middle right">
+                <p>Our records show that you’ve already seen this image. We might have run out of data for you in this workflow!</p>
+                <p>Try choosing a different workflow or contributing to a different project.</p>
+              </Tooltip>}
+          </button>
+
+        else if @props.subject.retired
+          <button type="button" className="warning-banner" onClick={@toggleWarning}>
+            Retired!
+            {if @state.showWarning
+              <Tooltip attachment="top left" targetAttachment="middle right">
+                <p>This subject already has enough classifications, so yours won’t be used in its analysis!</p>
+                <p>If you’re looking to help, try choosing a different workflow or contributing to a different project.</p>
```

Comment:
For sure, I'm just putting it off for the sake of getting stuff out there. I'm also not super happy with how this library's set up (overly complicated setup, no default fallback, weird interpolation syntax) so I'm going to shop around again before pulling all the strings out. I think Yahoo has something people like.